### PR TITLE
Add UnqualifiedSearchRegistries to the Image CR

### PR DESCRIPTION
**Pull request by mburke5678**
https://issues.redhat.com/browse/OSDOCS-1644

### CODE REVIEW

File: modules/images-configuration-allowed.adoc [hunk 1]
```diff
@@ -56,7 +56,7 @@ status:
 Either the `allowedRegistries` parameter or the `blockedRegistries` parameter can be set, but not both.
 ====
 +
-The Machine Config Operator (MCO) watches the `image.config.openshift.io/cluster` CR for any changes to registries and reboots the nodes when it detects changes. Changes to the allowed registries creates or updates the image signature policy in the `/host/etc/containers/policy.json` file on each node.
+The Machine Config Operator (MCO) watches the `image.config.openshift.io/cluster` CR for any changes to registries, then drains and uncordons the nodes when it detects changes. After the nodes return to the `Ready` state, the allowed registries list is used to update the image signature policy in the `/host/etc/containers/policy.json` file on each node.
```
Comment (mburke5678) [inline modules/images-configuration-allowed.adoc:59]: This is an editorial change for consistency only. No need for QE review.

File: modules/images-configuration-blocked.adoc [hunk 2]
```diff
@@ -49,7 +49,7 @@ status:
 Either the `blockedRegistries` registry or the `allowedRegistries` registry can be set, but not both.
 ====
 +
-The Machine Config Operator (MCO) watches the `image.config.openshift.io/cluster` CR for any changes to registries and reboots the nodes when it detects changes. Changes to the blocked registries appear in the `/etc/containers/registries.conf` file on each node.
+The Machine Config Operator (MCO) watches changes to the registries, then drains and uncordons the nodes when it detects changes. After the nodes return to the `Ready` state, changes to the blocked registries appear in the `/etc/containers/registries.conf` file on each node.
```
Comment (mburke5678) [inline modules/images-configuration-blocked.adoc:52]: This is an editorial change for 4.7 MCO change previously reviewed. No need for QE review.

File: modules/images-configuration-file.adoc [hunk 3]
```diff
@@ -42,7 +42,7 @@ spec:
       insecure: false
   additionalTrustedCA: <3>
     name: myconfigmap
-  registrySources:<4>
+  registrySources: <4>
```
Comment (mburke5678) [inline modules/images-configuration-file.adoc:45]: This is an editorial change for consistency only. No need for QE review.

File: modules/images-configuration-insecure.adoc [hunk 4]
```diff
@@ -60,7 +60,7 @@ status:
 When the `allowedRegistries` parameter is defined, all registries including the registry.redhat.io and quay.io registries are blocked unless explicitly listed. If you use the parameter, to prevent pod failure, add `registry.redhat.io` and `quay.io` to the `allowedRegistries` list, as they are required by payload images within your environment. 
 ====
 +
-The Machine Config Operator (MCO) watches the `image.config.openshift.io/cluster` CR for any changes to registries and reboots the nodes when it detects changes. Changes to the insecure and blocked registries appear in the `/etc/containers/registries.conf` file on each node.
+The Machine Config Operator (MCO) watches changes to the registries, then drains and uncordons the nodes when it detects changes. After the nodes return to the `Ready` state, changes to the insecure and blocked registries appear in the `/etc/containers/registries.conf` file on each node.
```
Comment (mburke5678) [inline modules/images-configuration-insecure.adoc:63]: This is an editorial change for consistency only. No need for QE review.

File: modules/images-configuration-parameters.adoc [hunk 5]
```diff
@@ -65,13 +65,17 @@ registries are allowed.
...
+other registries are blocked.
+
 Either `blockedRegistries` or `allowedRegistries` can be set, but not both.
 
 |===
 
 [WARNING]
 ====
-When the `allowedRegistries` parameter is defined, all registries including `registry.redhat.io` and `quay.io` are blocked unless explicitly listed. If using the parameter, declare source registries `registry.redhat.io` and `quay.io` as required by payload images within your environment, to prevent Pod failure. For disconnected clusters, mirror registries should also be added.
+When the `allowedRegistries` parameter is defined, all registries including `registry.redhat.io` and `quay.io` are blocked unless explicitly listed. If using the parameter, declare source registries `registry.redhat.io` and `quay.io` as required by payload images within your environment, to prevent pod failure. For disconnected clusters, mirror registries should also be added.
```
Comment (mburke5678) [inline modules/images-configuration-parameters.adoc:78]: This is an editorial change for consistency only. No need for QE review.

File: modules/images-configuration-parameters.adoc [hunk 6]
```diff
@@ -65,13 +65,17 @@ registries are allowed.
 `allowedRegistries`: Allowlisted for image pull and push actions. All other
 registries are blocked.
 
+`containerRuntimeSearchRegistries`: Allowlisted for image pull and push actions 
+for images that do not have fully qualified domains in their pull specs. All 
+other registries are blocked.
```
Comment (mburke5678) [inline modules/images-configuration-parameters.adoc:70]: QE, please review this change

File: modules/images-configuration-shortname.adoc [hunk 7]
```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * openshift_images/image-configuration.adoc
+// * post_installation_configuration/preparing-for-users.adoc
```
Comment (mburke5678) [inline modules/images-configuration-shortname.adoc:4]: QE, please review this new text.

**Comment by mburke5678** [comment]
@sunilcio PTAL. I made several editorial changes to ensure consistency throughout the other modules in the assembly. Please feel free to not review these changes.

**Comment by mburke5678** [comment]
@sunilcio PTAL

**Comment by sunilcio** [comment]
@mburke5678 thanks, lgtm!

File: modules/images-configuration-shortname.adoc [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+[WARNING]
+====
+Using image short names with public registries is strongly discouraged. You should use image short names with only internal or private registries.
+
+If you list public registries under the `containerRuntimeSearchRegistries` parameter, you expose your credentials to all the registries on the list and you risk network and registry attacks. You should always use fully-qualified image names with public registries.
+====
+
+The Machine Config Operator (MCO) watches the `image.config.openshift.io/cluster` resource for any changes to the registries. When the MCO detects a change, it drains the nodes, applies the change, and uncordons the nodes. After the nodes return to the `Ready` state, if the `containerRuntimeSearchRegistries` parameter is added, the MCO and creates a file in the `/etc/containers/registries.conf.d` directory on each node with the listed registries. The file overrides the default list of unqualified search registries in the `/host/etc/containers/registries.conf` file. There is no way to fall back to the default list of unqualified search registries.
```
Comment (codyhoag) [inline modules/images-configuration-shortname.adoc:24]:
```suggestion
The Machine Config Operator (MCO) watches the `image.config.openshift.io/cluster` resource for any changes to the registries. When the MCO detects a change, it drains the nodes, applies the change, and uncordons the nodes. After the nodes return to the `Ready` state, if the `containerRuntimeSearchRegistries` parameter is added, the MCO creates a file in the `/etc/containers/registries.conf.d` directory on each node with the listed registries. The file overrides the default list of unqualified search registries in the `/host/etc/containers/registries.conf` file. There is no way to fall back to the default list of unqualified search registries.
```

File: modules/images-configuration-shortname.adoc [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+The `containerRuntimeSearchRegistries` parameter works only with the Podman and CRI-O container engines and the registries in the list can be used only in pod specs, not in builds and image streams.
```
Comment (codyhoag) [inline modules/images-configuration-shortname.adoc:26]:
```suggestion
The `containerRuntimeSearchRegistries` parameter works only with the Podman and CRI-O container engines. The registries in the list can be used only in pod specs, not in builds and image streams.
```

File: modules/images-configuration-shortname.adoc [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+<1> Specify registries to use with image short names. You should use image short names with only internal or private registries to reduce possible security risks.
+<2> Ensure that any registries listed under `containerRuntimeSearchRegistries` are included in the `allowedRegistries` list.
++
+[NOTE]
+====
+When the `allowedRegistries` parameter is defined, all registries including the `registry.redhat.io` and `quay.io` registries are blocked unless explicitly listed. If you use this parameter, to prevent pod failure, add `registry.redhat.io` and `quay.io` to the `allowedRegistries` list, as they are required by payload images within your environment. 
+====
+
+To check that the registries have been added, use the following command on a node:
++
```
Comment (codyhoag) [inline modules/images-configuration-shortname.adoc:83]: The `+` markers are currently rendering (line 83 and 88). I'd suggest making this verification step a numbered step, which would fix the rendering issue.
> . To check that the registries have been added, use the following command on a node:

**Review by codyhoag — COMMENTED** [review]
Just a few suggestions; looks good otherwise! Just a reminder to add the labels before merging :slightly_smiling_face:

**Pull request merged**